apps/publisher/extensions/assets/default/apis/assets.jag

```diff
@@ -68,34 +68,45 @@ require('/modules/publisher.js').exec(function (ctx) {
             var newAsset;
             if (uriMatcher.match(CREATE_URL)) {//to create asset
                 try {
-                    newAsset = assetApi.create(options, req, res, session);
-                    //If the asset is successfully created then return the asset
-                    if (newAsset) {
-                        response = responseProcessor.buildSuccessResponse(res,
-                                constants.STATUS_CODES.CREATED, newAsset)
+                    var contentTypeheader = req.getHeader('Content-Type');
+                    var n = contentTypeheader.indexOf("multipart/form-data");
+                    if (contentTypeheader === 'application/json' || contentTypeheader === 'application/x-www-form-urlencoded' || n >= 0) {
```

Comment:
Again this condition check is not correct. i.e. A content type header might contain additional stuff separated by ;. Hence it might not exactly equal, even though it is application/json etc. Better to use indexOf or better implementation for that. 

Further, would be better to organize the code as below. So, it would be more readable.

if(!contentType) {
   //use a default content type or send an error
}

if(isContenType('applicaiton/json') {
    //call a method which does the application/json logic
    return;
}

if(isContentType('applicaiton/x-www-form-urlencoded')) {
    //call a method which does  applicaiton/x-www-form-urlencoded logic
    return;  
}

//send the error code saying content type is not supported.